jsonpb/jsonpb.go

```diff
@@ -974,11 +974,12 @@ func (u *Unmarshaler) unmarshalValue(target reflect.Value, inputValue json.RawMe
 		if mp != nil {
 			target.Set(reflect.MakeMap(targetType))
 			var keyprop, valprop *proto.Properties
+			//lint:ignore SA9003 These could still be nil if the
```

Comment:
Revert. Please don't add lint annotations to the source code.

Reply:
If we really want to silence the lint check, just comment out the if statement.

Reply:
Can you explain the objection? I was imagining that adding megacheck to CI would be the desired outcome here, and this comment is necessary for staticcheck-clean (which is a part of megacheck).

Reply:
Commenting out the check is slightly worse, since it'll allow the code to rot.

Anyway, tell me what you prefer and I'll update this when I get home. Any other comments?

Reply:
As a language, Go tends to favor less compiler and tooling hints and more of the tools "just doing the right thing". Of course, tools will never be able to determine programmer intent correctly all the time, so false-positives exists.

Some programmers are happy with tooling hints in their code and that's fine with them; it is really up to the owner of a given code base. In the protobuf repo, we don't want any tooling hints in our source code and that's a reasonable stance for us to hold as well.

Please make the change. Either comment out the if-statement or delete it entirely.

Reply:
Done.